### PR TITLE
Dala 2491 migrate p2 p to mldt

### PR DESCRIPTION
**Pull request by emanueldfine**
# Pull Request \

## Things to do during Peer Review
Please check all boxes before the Pull Request is merged. In case you skip a box, describe in the PRs description (that means: here) why the check is skipped.
- [x] The Github Actions (including Sonarqube Gateway and Lint Checks) are green. This is enforced by Github. 
- [x] A peer-review has been executed
  - [x] The code has been manually inspected by someone who did not implement the feature
- [x] The PR actually implements what is described in the JIRA-Issue
- [x] At least one test exists testing the new feature
  - [x] If you have created new test files, make sure that they are included in a test container and actually run in the CI
- [x] Documentation is updated as required
- [x] The automated deployment is updated if required
- [ ] If there was a database entity class added, there must also be a migration script for creating the corresponding database if flyway is already used by the service
- [ ] IF there are changes done to the framework data models or to a database entity class, the following steps were completed
  - [ ] The main branch is deployed to a dev server with `Reset non-user related Docker Volumes & Re-populate` turned on
  - [ ] It's verified that the CD run is green
  - [ ] Immediately after that the feature branch is deployed to the same server with `Reset non-user related Docker Volumes & Re-populate` turned off
  - [ ] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on the dev server
- [x] ELSE, the new version is deployed to the dev server "dev1" using this branch
  - [x] Run with setting `Reset non-user related Docker Volumes & Re-populate` turned on 
  - [x] It's verified that this version actually is the one deployed (check gitinfo for branch name and commit id!)
  - [x] It's verified that the CD run is green
  - [ ] The new feature is manually used/tested/observed on dev server
- [ ] Make sure that pull requests in the development tools repository, which are connected to this pull requests, 
are merged

### CODE REVIEW

File: dataland-frontend/src/components/resources/dataTable/MultiLayerDataTableCell.vue [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
 </template>
 
 <script lang="ts">
 import {
-  type AvailableDisplayValues,
-  MLDTDisplayComponents,
-} from "@/components/resources/dataTable/MultiLayerDataTableCells";
+  type AvailableMLDTDisplayObjectTypes,
+  MLDTDisplayComponentName,
+} from "@/components/resources/dataTable/MultiLayerDataTableCellDisplayer";
```
Comment (stewal19) [inline dataland-frontend/src/components/resources/dataTable/MultiLayerDataTableCell.vue:13]: That was just a renaming of the file, correct?

File: dataland-frontend/src/components/resources/dataTable/cells/DocumentLinkDisplayComponent.vue [hunk 2]
```diff
@@ -10,17 +10,17 @@
 <script lang="ts">
 import { defineComponent } from "vue";
 import {
-  type MLDTDisplayComponents,
-  type MLDTDisplayValue,
-} from "@/components/resources/dataTable/MultiLayerDataTableCells";
+  type MLDTDisplayComponentName,
+  type MLDTDisplayObject,
```
Comment (stewal19) [inline dataland-frontend/src/components/resources/dataTable/cells/DocumentLinkDisplayComponent.vue:14]: Thats also just renaming, correct?

File: dataland-frontend/src/components/resources/dataTable/MultiLayerDataTableCell.vue (hunk 1 above)
Comment (emanueldfine) [reply dataland-frontend/src/components/resources/dataTable/MultiLayerDataTableCell.vue:13]: Yes. I wanted to help other people understanding the MLDT better.
That's why I renamed everything that confused me during my time of digging into the MLDT. 
I hope it helps a little bit.

File: dataland-frontend/src/components/resources/dataTable/cells/DocumentLinkDisplayComponent.vue (hunk 2 above)
Comment (emanueldfine) [reply dataland-frontend/src/components/resources/dataTable/cells/DocumentLinkDisplayComponent.vue:14]: Yes.
I renamed `MLDTDisplayValue` to `MLDTDisplayObject` because it does not only contain the value to display, but also the info what name the component has that shall be used to display the value.
So I thought "MLDTDisplayObject" expresses this better.

File: dataland-frontend/tests/component/components/resources/dataTable/conversion/NumberValueGetterFactory.cy.ts [hunk 3]
```diff
@@ -1,41 +1,63 @@
...
   it("An empty string should be displayed if the data point is undefined", () => {
     const dataset = { data: undefined };
-    const value = numberValueGetterFactory("data")(dataset);
-    expect(value).to.deep.equal(EmptyDisplayValue);
+    const value = numberValueGetterFactory("data", sampleNumberFormFieldWithUnit)(dataset);
+    expect(value).to.deep.equal(MLDTDisplayObjectForEmptyString);
   });
 
-  it("The value of the input should be displayed if it exists", () => {
+  it("The value of the input should be displayed together with its unit if it exists", () => {
```
Comment (stewal19) [inline dataland-frontend/tests/component/components/resources/dataTable/conversion/NumberValueGetterFactory.cy.ts:28]: Do we need this test? We also check for the unit in the test below.

File: dataland-frontend/tests/component/components/resources/dataTable/conversion/NumberValueGetterFactory.cy.ts [hunk 4]
```diff
@@ -1,41 +1,63 @@
...
+      displayValue: "1,023 someUnit",
+    });
+  });
+
+  it("The value of the input should be displayed without a unit if it does not have one", () => {
+    const dataset = { data: 1234023.333 };
+    const value = numberValueGetterFactory("data", sampleNumberFormFieldWithoutUnit)(dataset);
+    expect(value).to.deep.equal(<MLDTDisplayObject<MLDTDisplayComponentName.StringDisplayComponent>>{
+      displayComponentName: MLDTDisplayComponentName.StringDisplayComponent,
+      displayValue: "1,234,023.33",
```
Comment (stewal19) [inline dataland-frontend/tests/component/components/resources/dataTable/conversion/NumberValueGetterFactory.cy.ts:60]: Is the unit not expected here?

File: dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTableWithP2p.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
+import { type FixtureData, getPreparedFixture } from "@sharedUtils/Fixtures";
+import { DataTypeEnum, type PathwaysToParisData } from "@clients/backend";
```
Comment (stewal19) [inline dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTableWithP2p.ts:2]: Is this the content of the old P2PViewPanel test?

File: dataland-frontend/src/components/resources/dataTable/cells/DocumentLinkDisplayComponent.vue (hunk 2 above)
Comment (stewal19) [reply dataland-frontend/src/components/resources/dataTable/cells/DocumentLinkDisplayComponent.vue:14]: ok, i ilke the changes. only wanted to make sure that nothing besides the naming was changed. i will resolve here

File: dataland-frontend/src/components/resources/dataTable/MultiLayerDataTableCell.vue (hunk 1 above)
Comment (stewal19) [reply dataland-frontend/src/components/resources/dataTable/MultiLayerDataTableCell.vue:13]: ok, i will resolve here

File: dataland-frontend/tests/component/components/resources/dataTable/conversion/NumberValueGetterFactory.cy.ts (hunk 4 above)
Comment (emanueldfine) [reply dataland-frontend/tests/component/components/resources/dataTable/conversion/NumberValueGetterFactory.cy.ts:60]: No because I use `sampleNumberFormFieldWithoutUnit` which says "WithoutUnit".

My goal is to assure that the number is still shown (without a unit attached to it) if the Field has no unit.

File: dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTableWithP2p.ts (hunk 5 above)
Comment (emanueldfine) [reply dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTableWithP2p.ts:2]: It is similar. However I have thrown out test code that makes no sense or is already covered by the respective DataIntegrity test of P2p.

Comment (emanueldfine) [reply dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTableWithP2p.ts:2]: (But yes, it is kind of a migration of the old P2pViewPanel test)

File: dataland-frontend/tests/component/components/resources/dataTable/conversion/NumberValueGetterFactory.cy.ts (hunk 3 above)
Comment (emanueldfine) [reply dataland-frontend/tests/component/components/resources/dataTable/conversion/NumberValueGetterFactory.cy.ts:28]: I haven't added this test, it already exists in main. But if you want my opinion on it:

Yes the unit is also checked in the test below. But the test below is even a little "stricter" because it additionally relies on the rounding to work as expected.

So if the unit is displayed correctly, but the rounding is broken, this test here will pass, but the one below will fail, telling you that the rounding must be broken.

Comment (stewal19) [reply dataland-frontend/tests/component/components/resources/dataTable/conversion/NumberValueGetterFactory.cy.ts:28]: ok

File: dataland-frontend/src/components/resources/frameworkDataSearch/p2p/P2pModalColumnHeaders.ts [hunk 6]
```diff
@@ -1,6 +1,6 @@
 export const p2pModalColumnHeaders = {
   driveMixPerFleetSegment: {
-    vehiclesType: "Vehicles type",
+    driveMixType: "Vehicles type",
```
Comment (stewal19) [inline dataland-frontend/src/components/resources/frameworkDataSearch/p2p/P2pModalColumnHeaders.ts:3]: Why this change?

File: dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTable.cy.ts [hunk 7]
```diff
@@ -158,7 +157,7 @@ describe("Tests for the MultiLayerDataTable component", () => {
       getCellContainer("Level 3 - String").should("be.visible");
 
       getSectionHead("Section 1").should("have.attr", "data-section-expanded", "true").click();
-      getSectionHead("Subsection 1").should("not.be.visible");
+      getSectionHead("Subsection 1", false).should("not.be.visible");
```
Comment (stewal19) [inline dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTable.cy.ts:160]: Here the repeating commands to check the section heads are moved to a function to get leaner code, correct?

File: dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTableWithP2p.ts (hunk 5 above)
Comment (stewal19) [reply dataland-frontend/tests/component/components/resources/dataTable/MultiLayerDataTableWithP2p.ts]: ok

File: dataland-frontend/src/components/resources/frameworkDataSearch/p2p/P2pModalColumnHeaders.ts (hunk 6 above)
Comment (stewal19) [reply dataland-frontend/src/components/resources/frameworkDataSearch/p2p/P2pModalColumnHeaders.ts:3]: Clarified in a call
